### PR TITLE
rancher-cifs volume driver for Azure file storage

### PR DESCRIPTION
**Pull request by mfar-bcgdv**

### CODE REVIEW

**Comment by mfar-bcgdv** [comment]
https://github.com/rancher/rancher-catalog/pull/612

**Comment by deniseschannon** [comment]
https://github.com/rancher/rancher/issues/9206

File: package/cifs/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,6 @@
+FROM ubuntu:16.04
+RUN apt-get update && \
+    apt-get install -y jq curl nfs-common
```
Comment (LLParse) [inline package/cifs/Dockerfile]: I had to replace `nfs-common` with `cifs-utils` to get this to work.

File: package/cifs/rancher-cifs [hunk 2]
```diff
@@ -0,0 +1,185 @@
+#!/bin/bash
```
Comment (LLParse) [inline package/cifs/rancher-cifs:1]: Can you please add executable permissions (`chmod +x`) to this file?

**Review by LLParse — CHANGES_REQUESTED** [review]

File: package/cifs/Dockerfile (hunk 1 above)
Comment (mfar-bcgdv) [reply package/cifs/Dockerfile]: hi @LLParse ,
I added `cifs-utils` to my Dockerfile based on Microsoft instructions:
https://docs.microsoft.com/en-us/azure/storage/storage-how-to-use-files-linux
This image been in use on some of our prod environments for couple months now without any hiccups.
let me know if you still wanna replace that package.

File: package/cifs/rancher-cifs (hunk 2 above)
Comment (mfar-bcgdv) [reply package/cifs/rancher-cifs:1]: done :)

File: package/cifs/Dockerfile (hunk 1 above)
Comment (LLParse) [reply package/cifs/Dockerfile]: To make sure I'm on the same page, you installed `cifs-utils` package on the host and the storage plugin functions, correct?

We typically try to ship all dependencies within the Docker image so the user doesn't need to perform any manual setup on the underlying hosts. In some scenarios (RancherOS, Atomic), installing deps on the host is not even possible. I would like to see the package added, thanks.

cc @moelsayed

File: package/cifs/rancher-cifs (hunk 2 above)
Comment (LLParse) [reply package/cifs/rancher-cifs:1]: Thanks!

File: package/cifs/Dockerfile (hunk 1 above)
Comment (mfar-bcgdv) [reply package/cifs/Dockerfile]: I used ubuntu vanilla 16:04 as my VM os & did not install any packages in os level.
Maybe `nfs-common` is already preinstalled in ubuntu 16.04.
lets add both of these packages in Dockerfile, agreed?

Comment (LLParse) [reply package/cifs/Dockerfile]: :+1:

Comment (mfar-bcgdv) [reply package/cifs/Dockerfile]: done.

**Comment by richbayliss** [comment]
Can we expect official CIFS support in the next release?

**Comment by LLParse** [comment]
@moelsayed  ^^

**Comment by richbayliss** [comment]
Is there any update on this one?

**Comment by annerajb** [comment]
Is this project dead?

**Comment by deniseschannon** [comment]
We've moved from using docker volumes to flex volumes. If this PR is re-worked to use flexvolumes, we can look to review and merge it in. 

PRs that were done for other storage drivers that moved to using flex volumes:
* https://github.com/rancher/storage/pull/58
* https://github.com/rancher/storage/pull/80
* https://github.com/rancher/agent/pull/229
* https://github.com/rancher/cattle/pull/2906

**Comment by cjellick** [comment]
Closing this due to staleness. Feel free to reopen or open a new PR if there's still a relevant change to be made. Thanks!

**Pull request closed without merging**